### PR TITLE
Change datasource url to proxy

### PR DESCRIPTION
**Pull request by QuentinBisson**
Follow up of https://github.com/giantswarm/loki-app/pull/352 this PR configures the new grafana-multi-tenant-proxy secrets to manage tenancy in front of the loki-gateway

### CODE REVIEW

**Review by QuantumEnigmaa — APPROVED** [review]
LGTM if tested

File: pkg/resource/proxy-auth/proxy-auth.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+		if err != nil {
+			return v1.Secret{}, errors.WithStack(err)
+		}
+
+		values.Users = append(values.Users, user{
+			Username: writeUser,
+			Password: writePassword,
+			// we set the tenant even though it may be given by the sender (promtail)
+			// depending of grafana-multi-teant-proxy config
+			Orgid: writeUser,
```
Comment (TheoBrigitte) [inline pkg/resource/proxy-auth/proxy-auth.go:85]: How does this work with keep org id setting ?

Comment (QuentinBisson) [reply pkg/resource/proxy-auth/proxy-auth.go:85]: You know I copied this from loki-auth to recreate the secret, I was not trying to change the implementation :D

File: pkg/resource/proxy-auth/proxy-auth.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+// listWriteUsers returns a map of users found in a credentialsSecret
+func listWriteUsers(credentialsSecret *v1.Secret) []string {
+	var usersList []string
+	for myUser := range credentialsSecret.Data {
+		// bypass read user
+		// bypass old creds (xxxuser and xxxpassword)
+
+		if !strings.HasSuffix(myUser, "user") && !strings.HasSuffix(myUser, "password") && myUser != common.ReadUser {
```
[the remark this answers is not in the document]
Comment (QuentinBisson) [reply pkg/resource/proxy-auth/proxy-auth.go:54]: That was old creds, I'm not sure if they exist anymore

File: pkg/resource/proxy-auth/proxy-auth.go (hunk 1 above)
Comment (QuentinBisson) [reply pkg/resource/proxy-auth/proxy-auth.go:85]: Well this is the enforced value for the user

File: pkg/resource/proxy-auth/proxy-auth.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+type Values struct {
+	Users []user `yaml:"users" json:"users"`
+}
+
+type user struct {
+	Username string `yaml:"username" json:"username"`
+	Password string `yaml:"password" json:"password"`
+	Orgid    string `yaml:"orgid" json:"orgid"`
+}
```
[the remark this answers is not in the document]
Comment (QuentinBisson) [reply pkg/resource/proxy-auth/proxy-auth.go:33]: This is the same structure yes. this operator actually creates this config. We could theoretically use the config value if it was public which might be a good idea to do later on

[the remark this answers is not in the document]
Comment (TheoBrigitte) [reply pkg/resource/proxy-auth/proxy-auth.go:33]: Alright, fine with me

File: pkg/resource/proxy-auth/proxy-auth.go (hunk 1 above)
Comment (QuentinBisson) [reply pkg/resource/proxy-auth/proxy-auth.go:85]: Yes

**Pull request merged**